src/realm/object-store/c_api/logging.cpp

```diff
@@ -0,0 +1,115 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or utilied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include <realm/object-store/c_api/types.hpp>
+#include <realm/object-store/c_api/util.hpp>
+
+namespace realm::c_api {
+namespace {
+using Logger = realm::util::Logger;
+static inline realm_log_level_e to_capi(Logger::Level level)
+{
+    switch (level) {
+        case Logger::Level::all:
+            return RLM_LOG_LEVEL_ALL;
+        case Logger::Level::trace:
+            return RLM_LOG_LEVEL_TRACE;
+        case Logger::Level::debug:
+            return RLM_LOG_LEVEL_DEBUG;
+        case Logger::Level::detail:
+            return RLM_LOG_LEVEL_DETAIL;
+        case Logger::Level::info:
+            return RLM_LOG_LEVEL_INFO;
+        case Logger::Level::warn:
+            return RLM_LOG_LEVEL_WARNING;
+        case Logger::Level::error:
+            return RLM_LOG_LEVEL_ERROR;
+        case Logger::Level::fatal:
+            return RLM_LOG_LEVEL_FATAL;
+        case Logger::Level::off:
+            return RLM_LOG_LEVEL_OFF;
+    }
```

Comment:
Ideally these would use the same numeric values so you could just do `return realm_log_level_e(level);`. At that point, you don't really need a function for it.

Reply:
So that's something we've done in the .NET SDK, but elsewhere the C API uses this pattern and I just followed it. To be honest I feel a bit queasy about just casting enum values like that because anyone can go ahead and change the numeric value of a member of `util::Logger::Level` without even realizing `realm_log_level_e` exists. I'd be all up for ditching the functions if we start generating the C API instead of maintaining it by hand, though.

Reply:
Couldn't you do the cast and turn these into a static assert instead? And add a comment where these two are defined that they must be in sync?

Reply:
That's not a bad idea at all! I'll give it a try.